File: Data Structures & Algorithms/subsets-ii/submission-0.java
class Solution {
    public List<List<Integer>> subsetsWithDup(int[] nums) {
        int n=nums.length;
        Arrays.sort(nums);
        Set<List<Integer>> l=new HashSet<>();
        for (int i=0;i<(1<<n);i++){
            List<Integer> m=new ArrayList<>();
            for(int j=0;j<n;j++){
                if((i&(1<<j))!=0){
                    m.add(nums[j]);
                }
            }
        l.add(m);
        }
        return new ArrayList<>(l);
    }
}

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        n=len(nums)
        li=[]
        for i in range(n):
            p=1
            for j in range(n):
                if (i==j):
                    continue
                else:
                    p=p*nums[j]
            li.append(p)
        return li
